modules/cert-manager-override-arguments.adoc

```diff
@@ -0,0 +1,69 @@
+// Module included in the following assemblies:
+//
+// * security/cert_manager_operator/cert-manager-customizing-api-fields.adoc
+
+:_content-type: PROCEDURE
+[id="cert-manager-override-arguments_{context}"]
+= Customizing cert-manager Operator API fields by overriding arguments
+
+You can override the default arguments for the {cert-manager-operator} by adding a `spec.controllerConfig` section in the `CertManager` resource.
+
+.Prerequisites
+
+* You have access to the {product-title} cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Edit the `CertManager` resource by running the following command:
++
+[source,terminal]
+----
+$ oc edit certmanager cluster
+----
+
+. Add a `spec.controllerConfig` section with the following override arguments:
++
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1alpha1
+kind: CertManager
+metadata:
+  name: cluster
+  ...
+spec:
+  ...
+  controllerConfig: <1>
+    overrideArgs:
+      - '--dns01-recursive-nameservers=<host>:<port>' <2>
+      - '--dns01-recursive-nameservers-only' <3>
+      - '--acme-http01-solver-nameservers=<host>:<port>' <4>
+      - '--v' <5>
+      - '-V' <5>
+      - '--metrics-listen-address=<host>:<port>' <7>
+----
+<1> Add the `spec.controllerConfig` section.
+<2> Provide a comma-separated list of `<host>:<port>` nameservers to query for the DNS-01 self check. For example, `--dns01-recursive-nameservers=1.1.1.1:53`.
+<3> Specify to only use recursive nameservers instead of checking the authoritative nameservers associated with that domain.
+<4> Provide a comma-separated list of `<host>:<port>` nameservers to query for the ACME HTTP01 self check. For example, `--acme-http01-solver-nameservers=1.1.1.1:53`.
+<5> Specify to set the log level verbosity to determine the amount of detail in logged error messages.
+<6> Specify the host and port for the metrics endpoint. The default value is `--metrics-listen-address=0.0.0.0:9402`.
+
+. Save the file to apply the changes.
+
+.Verification
+
+* Verify that the cert-manager pods are running by entering the following command:
++
+[source,terminal]
+----
+$ oc get pods -n cert-manager
```

Comment:
Same as the other one,
```
$ oc get pods -n cert-manager -o yaml
```

Reply:
I checked your recent change again wrt, but am leaning more towards the yaml output tbh, the rationale here being that in the verification step users would want to check if the args/env they set have been applied on the actual pod(s) or not.
I understand that the output is way too much, but can you snip it for relevant parts and still put it, i.e. eg.

```log
$ oc get pods -n cert-manager -o yaml
---
    name: cert-manager-6d4b5d4c97-kldwl
    namespace: cert-manager
---
  spec:
    containers:
    - args:
      - --v=2
      - --cluster-resource-namespace=$(POD_NAMESPACE)
      - --leader-election-namespace=kube-system
 ---
    name: cert-manager-cainjector-866c4fd758-ltxxj
    namespace: cert-manager
---
  spec:
    containers:
    - args:
      - --v=2
      - --leader-election-namespace=kube-system
---
    name: cert-manager-webhook-6d48f88495-c88gd
    namespace: cert-manager
---
  spec:
    containers:
    - args:
      - --v=2
      - --secure-port=10250
      - --dynamic-serving-ca-secret-namespace=$(POD_NAMESPACE)
      - --dynamic-serving-ca-secret-name=cert-manager-webhook-ca
      - --dynamic-serving-dns-names=cert-manager-webhook,cert-manager-webhook.$(POD_NAMESPACE),cert-manager-webhook.$(POD_NAMESPACE).svc
---
```